Update Modbus protocol service, adding MODBUS-TCP/IP mode.

pierrepitiot opened the pull request:
Remove org.eclipse.kura.protocol.modbus.test
Update examples/org.eclipse.kura.demo.modbus

This PR fixes issue #288 and #553 

@dwoodard1 Can you please review?

pierrepitiot committed:
Update Modbus protocol service, adding MODBUS-TCP/IP mode.

Remove org.eclipse.kura.protocol.modbus.test
Update examples/org.eclipse.kura.demo.modbus

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/OSGI-INF/metatype/org.eclipse.kura.demo.modbus.ModbusExample.xml:
```diff
@@ -74,6 +114,63 @@
+
+         <AD id="tcp.port"  
+            name="tcp.port"
+            type="String"
```
type should be integer with max 65535 and min 1 set - This would prevent users to provide alpha characters to the input field.

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/OSGI-INF/metatype/org.eclipse.kura.demo.modbus.ModbusExample.xml:
```diff
@@ -74,6 +114,63 @@
+
+        <AD id="pollInterval"  
+            name="pollInterval"
+            type="String"
```
type should be double

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/OSGI-INF/metatype/org.eclipse.kura.demo.modbus.ModbusExample.xml:
```diff
@@ -74,6 +114,63 @@
+
+        <AD id="publishInterval"  
+            name="publishInterval"
+            type="String"
```
type should be double

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/OSGI-INF/metatype/org.eclipse.kura.demo.modbus.ModbusExample.xml:
```diff
@@ -74,6 +114,63 @@
+            default="10" 
+            description="Interval in seconds between 2 publishes."/>
+
+        <AD id="inputAddress"  
```
type should be integer

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/OSGI-INF/metatype/org.eclipse.kura.demo.modbus.ModbusExample.xml:
```diff
@@ -74,6 +114,63 @@
+            default="" 
+            description="Address in decimal of the discrete input to read. Leave empty if no input."/>
+
+        <AD id="registerAddress"  
```
type should be integer

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/OSGI-INF/metatype/org.eclipse.kura.demo.modbus.ModbusExample.xml:
```diff
@@ -19,53 +19,93 @@
+            <Option label="256000" value="256000"/>
+            <Option label="460800" value="460800"/>
+            <Option label="921600" value="921600"/>
+		</AD>
```
There are some indentation issues in closing attribute definition tags.

amitjoy reviewed (changes requested):
Some metatype related improvements

pierrepitiot committed:
Modify Modbus demo metatype parameters for more input control.
Fix a bug in ModbusProtocolDevice (disconnecting Ethernet already
disconnected)
Remove org.eclipse.kura.protocol.modbus.test in manifest_pom.xml

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java:
```diff
@@ -44,99 +86,221 @@ public void unsetModbusProtocolDeviceService(ModbusProtocolDeviceService modbusS
+            if (this.m_properties.get(INPUT_ADDRESS) != null) {
+                inputaddr = (Integer) this.m_properties.get(INPUT_ADDRESS);
+            }
+            else inputaddr=0;
```
You don't need the else part as `inputaddr` is already initialised to 0. If there doesn't exist any property for `INPUT_ADDRESS`, it will always be 0.

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java:
```diff
@@ -44,99 +86,221 @@ public void unsetModbusProtocolDeviceService(ModbusProtocolDeviceService modbusS
+            if (this.m_properties.get(REGISTER_ADDRESS) != null) {
+                registeraddr = (Integer) this.m_properties.get(REGISTER_ADDRESS);
+            }
+            else registeraddr=0;
```
Same as before. Else part is not needed.

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java:
```diff
@@ -44,99 +86,221 @@ public void unsetModbusProtocolDeviceService(ModbusProtocolDeviceService modbusS
+            }
+
+            // schedule a new worker based on the properties of the service
+            int pubrate = 1000;
```
Instead of defaulting to 1000 in source code, you can add a default value to the `POLL_INTERVAL` in metatype.

amitjoy replied:
And it is already done in the metatype. So you don't need to initialise it to 1000.

pierrepitiot replied:
@amitjoy I think I have to set a default value if for some reason the property is null:
            int pubrate = 1000;
            if (this.m_properties.get(POLL_INTERVAL) != null) {
                pubrate = (Integer) this.m_properties.get(POLL_INTERVAL);
            }

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java:
```diff
@@ -149,68 +313,59 @@ private void configureDevice() throws ModbusProtocolException {
+        }
+        prop.setProperty("connectionType", modbusProtocol);
+
+        String Slave = "1";
```
Slave -> slave

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java:
```diff
@@ -223,26 +378,62 @@ private Properties getModbusProperties() {
+    // ----------------------------------------------------------------
+    @Override
+    public void onConnectionEstablished() {
+        // TODO Auto-generated method stub
```
If we want not to implement these methods, remove the TODO comments.  This is a code refactoring standard. You can do it like this.
`public void onConnectionEstablished() {}`

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java:
```diff
@@ -223,26 +378,62 @@ private Properties getModbusProperties() {
+
+    @Override
+    public void onConnectionLost() {
+        // TODO Auto-generated method stub
```
Same applies here

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java:
```diff
@@ -223,26 +378,62 @@ private Properties getModbusProperties() {
+    public void onMessagePublished(int arg0, String arg1) {
+        // TODO Auto-generated method stub
+
+    }
```
Same applies here to all these methods

amitjoy commented on kura/examples/org.eclipse.kura.demo.modbus/src/main/java/org/eclipse/kura/demo/modbus/ModbusExample.java:
```diff
@@ -149,68 +313,59 @@ private void configureDevice() throws ModbusProtocolException {
-    private Properties getModbusProperties() {
         Properties prop = new Properties();
 
+        String modbusProtocol = (String) this.m_properties.get(MODBUS_PROTOCOL);
```
An if condition will be helpful to check if the key is contained in the map or `this.m_properties.get(MODBUS_PROTOCOL) != null`.

amitjoy reviewed (changes requested):
Some minor code refactorings

pierrepitiot committed:
Some minor code refactorings

pierrepitiot commented:
Thak you @amitjoy for your vigilance :-)

amitjoy commented:
👍
